src/main/java/org/springframework/data/neo4j/repository/query/ReactivePartTreeNeo4jQuery.java

```diff
@@ -57,6 +57,10 @@ private ReactivePartTreeNeo4jQuery(ReactiveNeo4jOperations neo4jOperations, Neo4
 
 	static RepositoryQuery create(ReactiveNeo4jOperations neo4jOperations, Neo4jMappingContext mappingContext,
 			Neo4jQueryMethod queryMethod, ProjectionFactory factory) {
+		if (queryMethod.hasVectorSearchAnnotation()
+				&& queryMethod.getVectorSearchAnnotation().get().numberOfNodes() < 1) {
+			throw new IllegalArgumentException("Number of nodes in a vector search has to be greater than zero.");
```

Comment:
Nit: It makes sense to check how these exceptions translate during runtime and whether an enclosing exception provides sufficient context to identify the query method that has caused this verification error.